Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1099,11 +1099,14 @@ namespace Tensile
         granularities.suCuGranularity
             = granularities.suTilesPerCu / ceil(granularities.suTilesPerCu);
 
-        granularities.waveGranularity = std::min(
```

Comment:
cppcheck: Either the condition 'pAMDGPU!=nullptr' is redundant or there is possible null pointer dereference: pAMDGPU.
Indeed, there is a possible null pointer deference if pAMDGPU == nullptr. I've put a check before the pointer is used but I am not sure if this is correct either.

Reply:
Asserting or throwing an exception would be preferable.

Reply:
Added an assert to check if pAMDGPU is nullptr